clients/application/src/main/java/org/eclipse/hono/client/application/TelemetryMessage.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.client.application;
+
+import java.util.Objects;
+
+import org.eclipse.hono.util.QoS;
+
+import io.vertx.core.buffer.Buffer;
+
+/**
+ * A message of Hono's northbound Telemetry APIs.
+ */
+public class TelemetryMessage extends DownstreamMessage {
+
+    private final QoS qos;
```

Comment:
events also have a QoS and Ditto, for example, is interested in the QoS regardless of message type. So I guess we could/should pull this up into `AbstractDownstreamMessage` ...

Reply:
Good point. At that point, there is no difference in the implementation anymore. The messages are pure data holders. I'm not sure that we really need the hierarchy of interfaces then. We could remove `AbstractDownstreamMessage` as well as the interfaces `TelemetryMessage` and `EventMessage` and only have a class `DownstreamMessage` without losing anything. WDYT?

Reply:
I agree with `TelemetryMessage` and `EventMessage` being obsolete. However, IMHO we should still keep the `DownstreamMessage` interface. This should be helpful in unit tests as it can be mocked easily.